File: nagasenacs/src/Nagasena/Proc/IO/Compression/ScriberValueHolder.cs
using System.IO;

namespace Nagasena.Proc.IO.Compression {

  internal sealed class ScriberValueHolder {

    private int m_localName;
    private int m_uri;
    private int m_tp;

    private readonly ValueScriber m_valueScriber;
    private readonly object m_value;

    internal ScriberValueHolder(int localName, int uri, int tp, object value, ValueScriber valueScriber) {
      m_localName = localName;
      m_uri = uri;
      m_tp = tp;
      m_value = value;
      m_valueScriber = valueScriber;
    }

    internal void scribeValue(Stream outputStream, Scriber scriber) {
      m_valueScriber.doScribe(m_value, m_localName, m_uri, m_tp, outputStream, scriber);
    }

  }

}

File: sumerogi/src/com/sumerogi/proc/io/compression/ScriberValueHolder.java
package com.sumerogi.proc.io.compression;

import java.io.IOException;
import java.io.OutputStream;

import com.sumerogi.proc.io.Scriber;
import com.sumerogi.proc.io.ValueScriber;

final class ScriberValueHolder {

  private int m_localName;
  
  private final ValueScriber m_valueScriber;
  private final Object m_value;

  ScriberValueHolder(int localName, Object value, ValueScriber valueScriber) {
    m_localName = localName;
    m_value = value;
    m_valueScriber = valueScriber;
  }
  
  void scribeValue(OutputStream outputStream, Scriber scriber) throws IOException {
    m_valueScriber.doScribe(m_value, m_localName, outputStream, scriber);
  }

}
